File: Admin/models/plano.py
import re

class Plano:
    TEMPO_REGEX = re.compile(r"^\d+\s(ano|anos|mes|meses|semana|semanas)$")

    def __init__(self, id=0, nome="", valor=0.0, tempo=""):
        self.id = id
        self.nome = nome
        self.valor = valor
        self.tempo = tempo

    # Getters
    @property
    def id(self):
        return self.__id

    @property
    def nome(self):
        return self.__nome

    @property
    def valor(self):
        return self.__valor

    @property
    def tempo(self):
        return self.__tempo

    # Setters
    @id.setter
    def id(self, id: int):
        if not isinstance(id, int) or id < 0:
            raise ValueError("O ID deve ser um número inteiro positivo.")
        self.__id = id

    @nome.setter
    def nome(self, nome: str):
        if not isinstance(nome, str) or not nome.strip():
            raise ValueError("O nome do plano não pode estar vazio.")
        self.__nome = nome.strip()

    @valor.setter
    def valor(self, valor: float):
        if not isinstance(valor, (int, float)) or valor < 0:
            raise ValueError("O valor deve ser um número positivo.")
        self.__valor = float(valor)

    @tempo.setter
    def tempo(self, tempo: str):
        tempo = tempo.lower().replace("mês", "mes")  
        if not self.TEMPO_REGEX.match(tempo):
            raise ValueError('O tempo deve estar no formato "<número> <unidade>", exemplo: "1 ano", "12 meses", "2 semanas".')
        self.__tempo = tempo

    def to_dict(self):
        return {
            "id": self.id,
            "nome": self.nome,
            "valor": self.valor,
            "tempo": self.tempo
        }

    def __str__(self):
        return f"ID: {self.id}, Nome: {self.nome}, Valor: R$ {self.valor:.2f}, Tempo: {self.tempo})"

File: User/src/model/Plano.java
import java.util.HashMap;
import java.util.Map;
import java.util.regex.Pattern;

public class Plano implements Inter {
    private static final Pattern TEMPO_REGEX = Pattern.compile("^\\d+\\s(ano|anos|mes|meses|semana|semanas|dia|dias)$|^(ano|anos|mes|meses|semana|semanas|dia|dias)$");

    private int id;
    private String nome;
    private double valor;
    private String tempo;

    public Plano(int id, String nome, double valor, String tempo) {
        setId(id);
        setNome(nome);
        setValor(valor);
        setTempo(tempo);
    }

    public void setId(int id) {
        if (id < 0) {
            throw new IllegalArgumentException("O ID deve ser um numero inteiro positivo.");
        }
        this.id = id;
    }

    public void setNome(String nome) {
        if (nome == null || nome.trim().isEmpty()) {
            throw new IllegalArgumentException("O nome do plano nao pode estar vazio.");
        }
        this.nome = nome.trim();
    }

    public void setValor(double valor) {
        if (valor < 0) {
            throw new IllegalArgumentException("O valor do plano nao pode ser negativo.");
        }
        this.valor = valor;
    }

    public void setTempo(String tempo) {
        tempo = tempo.toLowerCase().replace("mês", "mes");

        if (!TEMPO_REGEX.matcher(tempo).matches()) {
            throw new IllegalArgumentException("O tempo de duracao nao eh valido.");
        }
        this.tempo = tempo;
    }

    public int getId() {
        return this.id;
    }

    public String getNome() {
        return this.nome;
    }

    public double getValor() {
        return this.valor;
    }

    public String getTempo() {
        return this.tempo;
    }

    public Map<String, Object> toMap() {
        Map<String, Object> map = new HashMap<>();
        map.put("id", id);
        map.put("nome", nome);
        map.put("valor", valor);
        map.put("tempo", tempo);
        return map;
    }

    @Override
    public String toString() {
        return "Plano(ID: " + this.id + ", Nome: " + this.nome + ", Valor: R$" + this.valor + ", Tempo: " + this.tempo + ")";
    }
}
